pyccel/parser/semantic.py

```diff
@@ -1109,11 +1146,34 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, new_expressions, is_augassign,ar
             if not arr_in_multirets:
                 self._ensure_target(rhs, d_lhs)
 
-            var = self.check_for_variable(name)
-
+            if isinstance(lhs, DottedName):
+                var = self.check_for_attribute(lhs)
+            else:
+                var = self.check_for_variable(name)
             # Variable not yet declared (hence array not yet allocated)
             if var is None:
-
+                if isinstance(lhs, DottedName):
+                    prefix = self._visit(DottedName(*lhs.name[:-1]))
+                    class_def = prefix.cls_base
+                    if isinstance(class_def, ClassDef) and lhs.name[0] == 'self':
+                        var      = self.get_variable('self')
+                        cls_name = str(var.cls_base.name)
+                        cls      = self.scope.find(cls_name, 'classes')
+
+                        name     = str(lhs.name[-1])
+                        name = self.scope.get_expected_name(name)
+                        member = self._create_variable(name, dtype, rhs, d_lhs)
+
+                        lhs    = member.clone(member.name, new_class = DottedVariable, lhs = var)
+                        if not str(lhs.name[-1]) == name:
+                            self.scope.insert_variable(lhs, name)
```

Comment:
Why does this variable only need inserting if the name has changed? Doesn't it always need inserting?

Reply:
It s already insterted (I insert just it just if the `get_expected_name` returns a new name)

Reply:
I added the `self.scope.remove_variable` to delete the previous variable from the scope.

Reply:
> It s already insterted

Can you point me at the code where this happens please

Reply:
the `id(var)` in here equals the `id(lhs)` in the `__assign_lhs_variable()` 
https://github.com/pyccel/pyccel/blob/275b85717e8bfc2f42eab6646ec3007f3753aa09/pyccel/parser/semantic.py#L3270-L3271

Reply:
Huh, how does that happen?
In:
```python
class Point:
    def __init__(self : Point, x : float):
        self._x = x
```
Don't we have `var`, which is `self`, and `lhs` which is `self._x`?

Reply:
and I just investigated and the problem comes from:
https://github.com/pyccel/pyccel/blob/1409206f5f7e43db2ba26bdbf971a9d81c9dd74c/pyccel/parser/semantic.py#L3342-L3343

Reply:
Fixed https://github.com/pyccel/pyccel/pull/1484/commits/802d795332b4c5c772830c7bd197c4b16347f873.